### PR TITLE
updated logic for endNodePattern to include label information in the variable name

### LINKED ISSUE #295: Wrong node name used for end node in rich relationship
**Description**
In our schema we have defined rich relationship. When we are trying to filter on the end node of the relationship, it is throwing error.

***GraphQL schema***

```
type Person{
    name: String
    age: Int
    rel_has_target_book: [Person_HAS_Target_Book]
}

type Book{
    name: String
    price: Int
}

type Person_HAS_Target_Book @relation(name: "HAS", from: "source", to: "target", direction: OUT) {
    name: String
    source: Person
    target: Book
}
```

***GraphQL request***

```
query{
  person{
    name
    rel_has_target_book(where:{target:{name:"Book 1"}}){
      target{
        name
      }
    }
  }
}
```

***Generated Cypher query***
Here, "personRel_has_target_bookTarget" is the variable name for Book but in the where condition, "personRel_has_target_bookBook" is being used
```
MATCH (person:Person)
CALL {
    WITH person
    MATCH (person)-[personRel_has_target_book:HAS]->(personRel_has_target_bookTarget:Book)
    WHERE personRel_has_target_bookBook.name = $personRel_has_target_bookBookName
    RETURN collect(personRel_has_target_book {
        target: personRel_has_target_bookTarget {
            .name
        }
    }) AS personRel_has_target_book
}
RETURN person {
    .name,
    rel_has_target_book: personRel_has_target_book
} AS person

Variables: {personRel_has_target_bookBookName=Book 1}
```

***Expected cypher query***

```
MATCH (person:Person)
CALL {
    WITH person
    MATCH (person)-[personRel_has_target_book:HAS]->(personRel_has_target_bookBook:Book)
    WHERE personRel_has_target_bookBook.name = $personRel_has_target_bookBookName
    RETURN collect(personRel_has_target_book {
        target: personRel_has_target_bookBook {
            .name
        }
    }) AS personRel_has_target_book
}
RETURN person {
    .name,
    rel_has_target_book: personRel_has_target_book
} AS person

Variables: {personRel_has_target_bookBookName=Book 1}
```

**Additional context**
We looked into the codebase. 
In the file "**ProjectionBase.kt**",  if we modify the below line in the function "**projectRichAndRegularRelationship**", then the correct cypher query is being generated.

Before: node(label).named("$childVariable${**relInfo.endField**.capitalize()}") to **relInfo.endField**

After: node(label).named("$childVariable${**label**.capitalize()}") to **label**

### PR DESCRIPTION
**Pull request by saurav213**
updated logic for endNodePattern to include label information in the variable name.

It fixes #295

### CODE REVIEW

**Comment by Andy2003** [comment]
Can you add a test that coveres the bug  (with the wrong where-parameters)

**Comment by saurav213** [comment]
Added a test case to validate variable name in where condition.

File: core/src/test/resources/filter-tests.adoc [hunk 1]
```diff
@@ -72,7 +86,24 @@ UNWIND [{
 AS props
 
 CREATE (p:Person)-[:WORKS_AT]->(c)
-SET p = props
+SET p = props;
```
Comment (Andy2003) [inline core/src/test/resources/filter-tests.adoc:89]: why adding test data, if you don't use it in the tests?

Comment (saurav213) [reply core/src/test/resources/filter-tests.adoc:89]: In below tests, there are graphql queries which are filtering on node properties. So this "props" is required here.

But in my test case for "wrong variable name in where condition", props is not required. I will remove it from there.

Comment (saurav213) [reply core/src/test/resources/filter-tests.adoc:89]: Removed unwanted data from my test case

Comment (Andy2003) [reply core/src/test/resources/filter-tests.adoc:89]: What I meant is that if you add test data, which is perfectly OK, then the test should also check if a particular query returns that data. This is tested using the following block:

```adoc
.GraphQL-Response
[source,json,response=true,ignore-order]
----
{
 "expected": "graphql response"
}
----
```

**Review by Andy2003 — COMMENTED** [review]
Do you want to add integration test assertions?

**Comment by saurav213** [comment]
Hi.
I will add it today. I got held up with some other work.

File: core/src/test/resources/filter-tests.adoc (hunk 1 above)
Comment (saurav213) [reply core/src/test/resources/filter-tests.adoc:89]: Added Graphql Response

**Comment by saurav213** [comment]
> Do you want to add integration test assertions?


You are talking about adding this response in my test, right?

```
.GraphQL-Response
[source,json,response=true,ignore-order]
----
{
 "expected": "graphql response"
}
----
```

**Comment by Andy2003** [comment]
Please fix the tests!

**Comment by saurav213** [comment]
> Please fix the tests!

In my local, all the CypherTests are running fine.

picc

After running mvn clean install, I don't see any test failure in Neo4j GraphQL Java 1.7.1-SNAPSHOT 

image

**Comment by Andy2003** [comment]
I will take a look at this

**Pull request closed without merging**